### DISCUSSION TITLE
Understanding Emoji / Unicode Rendering in Terminal

### DISCUSSION THREAD

**Discussion by MartinSGill** [microsoft/terminal#13724] [Q&A]
I'm trying to understand what the current situation is with regards to Emoji/Unicode support in Terminal, in a simple straightforward manner for people (like me) without detailed knowledge of fonts/Unicode or how DirectDraw works.

I found many issues (most closed) discussing aspects of this, but I'm finding it really hard to read any concrete answers in them.

Perhaps a (very) short explanation of how Windows displays emoji might help me. I read somewhere that this is done by DirectDraw APIs. Does DirectDraw detect the Emoji sequence and instead of drawing the font "letter" it instead renders a "colour image" for that sequence?

With reference to the screenshot at the bottom, what I'd really like to know:

1. Is this something I can fix? (e.g. a setting, different font)
2. Is this something I should report? (and to whom)

---

The **tiny** symbols. Is that an issue with the font, a setting I need to adjust in Windows, a setting I need to tweak in Terminal, or just nothing to be done?

Issue https://github.com/microsoft/terminal/issues/900 talks about this, with @DHowett [mentioning this](https://github.com/microsoft/terminal/issues/900#issuecomment-822577068):
> [...] The default representation for ⚠ is "narrow, single-width, non-emoji". The bug here is that it is yellow (emoji presentation), not that it is _small_.

I notice that all my "tiny" symbols are marked as "unqualified" in Unicode. I have no idea what that really means, but
is there a way for me to change the "default representation" for all "unqualified" Unicode emojis so they are the same size as other emojis?

---

The **missing** symbols. Is that a problem with my font not supporting the character, or with Windows not supporting that character (Unicode version?) Do I need to change my terminal font, or wait for an OS update?

---

![image](https://user-images.githubusercontent.com/2271740/184141668-44109c1b-bcfd-4bd5-aa86-3e9634e62923.png)

**Comment by devarda** [comment]
Oh man, if you get into the weeds of it the font rendering on Terminal is really hard at first... Until you figure it out.
